### DISCUSSION TITLE
non-english characters file name become gibberish when using .getEntries()

### DISCUSSION THREAD

**Discussion by chaintng** [gildas-lormeau/zip.js#542] [Q&A] [answered]
I have this zip file: https://drive.google.com/file/d/1v6s5iOkfFEzKRH-yeyS0qqMYL7fxHOYs/view?usp=share_link

It contain html and png files as following
(Those are Thai language characters)

```
 ✘  ~/Downloads/only-cookings  find . -print | sed -e 's;[^/]*/;|____;g;s;____|; |;g'

.
|____.DS_Store
|____Cooking 0462adc408af4ec7b9c1fadda6bca3e7
| |____.DS_Store
| |____ดำรงค์เจริญค้ากุ้ง กุ้ง กุ้งสด shrimp prawn ea607ec550fb4ad38e3eacee435b25ff
| | |____0731E892-0D33-4164-B702-660C1BD933CF.png
| |____ดำรงค์เจริญค้ากุ้ง กุ้ง กุ้งสด shrimp prawn ea607ec550fb4ad38e3eacee435b25ff.html
```

<img width="692" alt="image" src="https://github.com/user-attachments/assets/2480ca50-531d-445c-a33b-d697b54aef0f">

However, when using `zip.getEntries()` to this file,
I got gibberish

```
only-cookings/Cooking 0462adc408af4ec7b9c1fadda6bca3e7/α╕öα╕│α╕úα╕çα╕äα╣îα╣Çα╕êα╕úα╕┤α╕ìα╕äα╣ëα╕▓α╕üα╕╕α╣ëα╕ç α╕üα╕╕α╣ëα╕ç α╕üα╕╕α╣ëα╕çα╕¬α╕ö shrimp prawn ea607ec550fb4ad38e3eacee435b25ff.html
```

![image](https://github.com/user-attachments/assets/59c23254-6c8b-442e-8b76-62fad4ceb352)

I'm not sure what cause this

**Comment by 0f-0b** [comment, marked as answer]
The filenames in the archive are encoded in UTF-8, but the language encoding flag that indicates such isn't set, so zip.js instead tries to decode the filenames as if they are encoded in [CP437](https://en.wikipedia.org/wiki/Code_page_437), resulting in gibberish. This problem was discussed in detail in #131.

The easiest way to fix this is to set the [`filenameEncoding`](https://gildas-lormeau.github.io/zip.js/api/interfaces/GetEntriesOptions.html#filenameEncoding) option to `"utf-8"` ([table of all supported values](https://encoding.spec.whatwg.org/#concept-encoding-get)). This tells zip.js to decode the filenames as UTF-8 even in the absence of the language encoding flag. For more control over how the filenames are decoded, you can use the [`decodeText`](https://gildas-lormeau.github.io/zip.js/api/interfaces/GetEntriesOptions.html#decodeText) option or read the [`rawFilename`](https://gildas-lormeau.github.io/zip.js/api/interfaces/Entry.html#rawFilename) property.

**Reply by gildas-lormeau** [reply]
I should have opened the zip file... Thank you!

**Reply by chaintng** [reply]
Thank you for explanation.
Turn out, it seem to be a problem only the zip file that zip via cli in MacOS

I used this cli
```
zip -r test.zip ./*
```

I redownload zip filed from Notion, and this time, make zip file untouched (earlier, I expanded and zip it again on my machine)

It's no longer have the issue.

I think for some reason, MacOS encoded the file name in unexpected ways

Anyway, thanks for workaround 🙏
